src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4408,18 +4406,56 @@ void setAnalyzedNullabilityAsContinuation(
                 };
             }
 
-            static Symbol? getTargetMember(TypeSymbol containingType, BoundObjectInitializerMember objectInitializer)
+            Symbol? getTargetMember(TypeSymbol containingType, BoundObjectInitializerMember objectInitializer)
             {
                 var symbol = objectInitializer.MemberSymbol;
+                if (symbol == null)
+                    return null;
```

Comment:
i assume this is an error scenario?  in those cases, is it worth documenting/asserting that some diagnostic has already been reported in some fashion?  I'm not familiar with general compiler patterns, and if that would be overkill or not.

Reply:
Opened #80030 to try and answer this.

I was surprised to find that entirely commenting out the null check didn't fail any NullableReferenceTypesTests or a few other test classes that I tried locally.

What I know is that `.MemberSymbol` was annotated as nullable 6 years ago. I don't know if that was supposed to mean an error case or something else. We'll see with a little more investigation.

Reply:
It looks like one of the main cases where this is null, is when dynamic is being used. For example, `new C { [x] = y }`, when `x` is dynamic, an overload resolution needs to be done at runtime to decide which indexer is being used.

Basically it looks like this means there's not something wrong with this `objectInitializer`, but, we don't know what symbol is being used at compile time. I can add a comment in a separate PR.

See [test results](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1131427&view=ms.vss-test-web.build-test-results-tab&runId=31127412&resultId=143280&paneView=debug).

Reply:
i'm happy with this if you try to add a test for this.

Reply:
I added a simple test in NullableReferenceTypesTests which crashes if we don't properly handle a null MemberSymbol. I think that's all for us to do here.